modules/sdp/SdpSimulcast.ts

```diff
@@ -0,0 +1,313 @@
+                primarySsrc = Number(media.ssrcs[0]?.id);
+            } else {
+                const fidGroup = media.ssrcGroups.find(group => group.semantics === 'FID');
+
+                if (fidGroup) {
+                    primarySsrc = Number(fidGroup.ssrcs.split(' ')[0]);
+                }
+            }
+
+            if (this._ssrcCache.has(mid)) {
+                media = this._fillSsrcsFromCache(media);
+            } else {
+                media = this._generateNewSsrcsForSimulcast(media, primarySsrc);
+                const simulcastSsrcs = this._parseSimLayers(media);
+
+                // Update the SSRCs in the cache so that they can re-used for the same mid again.
+                this._ssrcCache.set(mid, simulcastSsrcs);
+            }
+        }
+
+        return new RTCSessionDescription({
+            type: description.type,
+            sdp: transform.write(session)
+        });
+    }
+
+    /**
+     * Munges the given media description by removing the SSRCs and related FID groups for the higher layer streams.
+     *
+     * @param description
+     * @returns
+     */
+    mungeRemoteDescription(description: Description) : Description {
+        if (!description || !description.sdp) {
+            return description;
+        }
+
+        const session = transform.parse(description.sdp);
+
+        for (const media of session.media) {
+            if (media.type !== MediaType.VIDEO) {
+                continue;
+            }
+
+            if (media.direction !== MediaDirection.SENDONLY) {
+                continue;
+            }
+
+            // Ignore m-lines that do not have any SSRCs or SSRC groups. These are the ones associated with remote
+            // sources that have left the call. These will be recycled when a new remote source joins the call.
+            if (!media.ssrcGroups?.length || !media?.ssrcs.length) {
+                continue;
+            }
+
+            // Cache the SSRCs and the source groups.
+            const mungedSsrcs = new Set(media.ssrcs.slice());
+            const mungedSsrcGroups = new Set(media.ssrcGroups.slice());
+            const fidGroups = media.ssrcGroups.filter(group => group.semantics === 'FID');
+            const simGroup = media.ssrcGroups.find(group => group.semantics === 'SIM');
+            const primarySsrc = simGroup?.ssrcs.split(' ')[0];;
+
+            // When simulcast and RTX are both enabled.
+            if (fidGroups.length && simGroup) {
+                const fidGroup = fidGroups.find(group => group.ssrcs.includes(primarySsrc));
+                const secondarySsrc = fidGroup.ssrcs.split(' ')[1];
+
+                for (const ssrcGroup of media.ssrcGroups) {
+                    if (ssrcGroup !== fidGroup) {
+                        mungedSsrcGroups.delete(ssrcGroup);
+                    }
+                }
+                for (const ssrc of media.ssrcs) {
+                    if (ssrc.id.toString() !== primarySsrc
```

Comment:
seems like the number is preferred overall, so maybe `primarySsrc` can be converter via `Number()` instead of `toString()` on the `id`?

Reply:
the same applies to `secondarySsrc`

Reply:
The inferred type for `primarySsrc` and `secondarySsrc` is `string` since they are derived from the media.ssrcGroups and hence I am doing it the other way.

Reply:
Right, but if they were wrapped in `Number` then it would be more consistent to work on the numbers only. Anyway not a big deal.